bot/seasons/evergreen/bookmark.py

```diff
@@ -45,21 +50,75 @@ def __init__(self, bot: commands.Bot):
 
+        error_embed = discord.Embed(
+            title=random.choice(ERROR_REPLIES),
+            description=f"Please enable DMs to receive the bookmark. "
+                        f"Once done you can retry by reacting with {Emojis.pin}",
+            colour=Colours.soft_red
+        )
+
         try:
             await ctx.author.send(embed=embed)
         except discord.Forbidden:
-            error_embed = discord.Embed(
-                title=random.choice(ERROR_REPLIES),
-                description=f"{ctx.author.mention}, please enable your DMs to receive the bookmark",
-                colour=Colours.soft_red
-            )
-            await ctx.send(embed=error_embed)
+            await ctx.send(embed=error_embed, delete_after=7.5)
+            sent_person = set()
         else:
-            log.info(f"{ctx.author} bookmarked {target_message.jump_url} with title '{title}'")
             await ctx.message.add_reaction(Emojis.envelope)
+            sent_person = {ctx.author}  # set of id who got the message
+
+        log.info(f"{ctx.author} bookmarked {target_message.jump_url} with title '{title}'.")
+        await ctx.message.add_reaction(Emojis.pin)
+
+        copied_embed = deepcopy(embed)
+
+        copied_embed.add_field(
+            name=f'Bookmarked from {ctx.author.name}.',
+            value=f'[Visit original message]({ctx.message.jump_url})',
+            inline=False
+        )
+
+        def check(reaction: discord.Reaction, user: discord.User) -> bool:
+            return (
+                user != self.bot.user
+                and reaction.emoji == Emojis.pin
+                and reaction.message == ctx.message
+                and user not in sent_person
+            )
+
+        while True:
+            try:
+                reaction, user = await self.bot.wait_for("reaction_add", timeout=60.0, check=check)
+            except asyncio.TimeoutError:
+                await ctx.message.clear_reactions()
+                return
+
+            log.info(f"{user.name} bookmarked {target_message.jump_url} with title '{title}' from '{ctx.author}'.")
+
+            try:
+                if user == ctx.author:
+                    await user.send(embed=embed)
+                else:
+                    await user.send(embed=copied_embed)
+
+            except discord.Forbidden:
+                await ctx.send(f"{user.mention} Please enable your DM to receive the message.", delete_after=7.5)
+                try:
+                    await reaction.remove(user)
+                except discord.Forbidden:
+                    await ctx.send("I don't have permissions to remove reaction.", delete_after=7.5)
+                except discord.HTTPException:
+                    await ctx.send("Unknown error while removing reaction.", delete_after=7.5)
+                except discord.NotFound:
+                    await ctx.send("Message or reaction author not found, can't remove reaction.", delete_after=7.5)
```

Comment:
This is caught by pylint:

![image](https://user-images.githubusercontent.com/19921765/77986915-82efa280-7342-11ea-97ae-4ba5bc3a7829.png)

Which makes sense, because `NotFound` is a subclass of `HTTPException` - Consider changing the order of catching.